lunatrace/bsl/backend/src/github/actions/pr-comment-generator.ts

```diff
@@ -67,7 +71,137 @@ function generatePullRequestCommentFromReport(projectId: string, scan: InsertedS
   return messageParts.join('\n');
 }
 
-export async function commentOnPrIfExists(buildId: string, scanReport: InsertedScan) {
+async function executePRComment(
+  buildLookup: GetBuildQuery,
+  scanReport: InsertedScan,
+  buildId: string,
+  projectId: any,
+  body: string,
+  installationId: number,
+  pullRequestId: string,
+  previousReviewId: string | null
+) {
+  const githubClient = await generateGithubGraphqlClient(installationId);
+  if (githubClient.error) {
+    log.error(`unable to create github client`, {
+      projectId,
+      installationId,
+      pullRequestId,
+    });
+    return;
+  }
+  const github = githubClient.res;
+
+  // This is the first build on this pr so make a new comment
+  if (!previousReviewId) {
+    const githubReviewResponse = await github.AddPrReview({
+      pull_request_id: pullRequestId.toString(),
+      body,
+    });
+    const existing_github_review_id = githubReviewResponse.addPullRequestReview?.pullRequestReview?.id;
+    if (!existing_github_review_id) {
+      return log.error('Failed to generate a review on pr, github responded ', githubReviewResponse);
+    }
+
+    log.info('review created');
+
+    await hasura.UpdateBuildExistingReviewId({ id: buildId, existing_github_review_id });
+    return;
+  }
+
+  // Otherwise just update the existing review on the PR.  Very similar to above but we update instead
+  const githubReviewResponse = await github.UpdatePrReview({
+    pull_request_review_id: previousReviewId,
+    body,
+  });
+  const existing_github_review_id = githubReviewResponse.updatePullRequestReview?.pullRequestReview?.id;
```

Comment:
Nit: camelCase for variables in JS

Reply:
(we should make the linter do this)

Reply:
I did this so I could inline it below, external api expects it this way.